### PR TITLE
[WIP] ECO-1221 Fixing null pointer exception in new build

### PR DESCRIPTION
**Pull request by moizjv**
@halkeye very initial version of working plugin, I still need to clean up. But root cause seem to be build.getAction(SauceOnDemandBuildAction.class) is null.

### CODE REVIEW

File: src/main/java/hudson/plugins/sauce_ondemand/SauceOnDemandBuildAction.java [hunk 1]
```diff
@@ -102,10 +102,24 @@ protected SauceCredentials getCredentials() {
...
     }
 
+    /**
+     * @param build The build in progress
+     * @return the {@link SauceOnDemandBuildAction} instance which has been registered with the build
+     *         Can be null
+     */
+    private static SauceOnDemandBuildAction getBuildAction(Run build) {
+        SauceOnDemandBuildAction buildAction = build.getAction(SauceOnDemandBuildAction.class);
+        if (buildAction == null && build instanceof hudson.maven.MavenBuild) {
```
Comment (halkeye) [inline src/main/java/hudson/plugins/sauce_ondemand/SauceOnDemandBuildAction.java]: Lets import MavenBuild class so its consistent

But yea, I can't wait till they remove MavenBuild. I'm not sure thats ever happening though.

File: src/main/java/hudson/plugins/sauce_ondemand/SauceOnDemandReportPublisher.java [hunk 2]
```diff
@@ -281,7 +281,7 @@ private void processBuildOutput(Run build, SauceOnDemandBuildAction buildAction,
     }
 
     protected SauceREST getSauceREST(Run build) {
-        SauceCredentials credentials = build.getAction(SauceOnDemandBuildAction.class).getCredentials();
+        SauceCredentials credentials = getBuildAction(build).getCredentials();
```
Comment (halkeye) [inline src/main/java/hudson/plugins/sauce_ondemand/SauceOnDemandReportPublisher.java]: The function is SauceOnDemandBuildAction, so this won't work will it?

Does anywhere look good to share it?

Comment (moizjv) [inline src/main/java/hudson/plugins/sauce_ondemand/SauceOnDemandReportPublisher.java]: It calls below function, and it works, I am still wrapping my head around what exactly is happening here. What do you mean by sharing in this context?

``` java
private SauceOnDemandBuildAction getBuildAction(Run build) {
        SauceOnDemandBuildAction buildAction = build.getAction(SauceOnDemandBuildAction.class);
        if (buildAction == null && build instanceof MavenBuild) {
            //try the parent
            buildAction = ((MavenBuild) build).getParentBuild().getAction(SauceOnDemandBuildAction.class);
        }
        return buildAction;
    }
```

**Comment by moizjv** [comment]
Oops wrong directory.

**Pull request closed without merging**